packages/vite/src/node/importGlob.ts

```diff
@@ -38,27 +53,49 @@ export async function transformImportGlob(
   importer = cleanUrl(importer)
   const importerBasename = path.basename(importer)
 
-  let [pattern, endIndex] = lexGlobPattern(source, pos)
-  if (!pattern.startsWith('.') && !pattern.startsWith('/')) {
-    throw err(`pattern must start with "." or "/" (relative to project root)`)
+  const [userPattern, endIndex] = lexGlobPattern(source, pos)
+
+  let globParams: {
+    base: string
+    pattern: string
+    parentDepth: number
+    isAbsolute: boolean
+  } = {
+    base: '',
+    pattern: userPattern,
+    parentDepth: 0,
+    isAbsolute: false
   }
```

Comment:
I think we can extract this as an interface so it can be used by the `formatGlobRelativePattern` function return type too.

And maybe we can have this as undefined by default as well so on line 93 below, we can check `if (!globParams)` instead.

So all in all, something like `let globParams: GlobParams | undefined = undefined`